src/main/java/seedu/address/logic/parser/EditRestaurantNoteCommandParser.java

```diff
@@ -0,0 +1,104 @@
+import static seedu.address.logic.parser.CliSyntax.PREFIX_LINE_NUMBER_BAD;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_LINE_NUMBER_GOOD;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_LINE_NUMBER_RECOMMENDED;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_RECOMMENDED;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.logic.commands.EditRestaurantNoteCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.restaurant.Note;
+
+/**
+ * Parses input arguments and creates a new {@code EditInfoCommand} object
+ */
+public class EditRestaurantNoteCommandParser implements Parser<EditRestaurantNoteCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the {@code EditRestaurantNoteCommand}
+     * and returns a {@code EditRestaurantNoteCommand} object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public EditRestaurantNoteCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args,
+                PREFIX_LINE_NUMBER_RECOMMENDED, PREFIX_RECOMMENDED, PREFIX_LINE_NUMBER_GOOD, PREFIX_GOOD,
+                PREFIX_LINE_NUMBER_BAD, PREFIX_BAD);
+
+        Index index;
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+        } catch (IllegalValueException ive) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    EditRestaurantNoteCommand.MESSAGE_USAGE), ive);
+        }
+
+        if (!argMultimap.getValue(PREFIX_LINE_NUMBER_RECOMMENDED).isPresent()
+                && !argMultimap.getValue(PREFIX_LINE_NUMBER_GOOD).isPresent()
+                && !argMultimap.getValue(PREFIX_LINE_NUMBER_BAD).isPresent()) {
+            throw new ParseException(String.format(MESSAGE_NO_LINE_NUMBER_RESTAURANT,
+                    EditRestaurantNoteCommand.MESSAGE_USAGE));
+        }
+
+        int lineRec = -1;
+        String recommendedFood = "";
+        if (argMultimap.getValue(PREFIX_LINE_NUMBER_RECOMMENDED).isPresent()) {
+            if (argMultimap.getAllValues(PREFIX_LINE_NUMBER_RECOMMENDED).get(0).isEmpty()) {
+                throw new ParseException(String.format(MESSAGE_NO_LINE_NUMBER_REC,
+                        EditRestaurantNoteCommand.MESSAGE_USAGE));
+            } else if (!argMultimap.getValue(PREFIX_RECOMMENDED).isPresent()
+                    || (argMultimap.getValue(PREFIX_RECOMMENDED).isPresent()
+                    && argMultimap.getAllValues(PREFIX_RECOMMENDED).get(0).isEmpty())) {
+                throw new ParseException(EditRestaurantNoteCommand.MESSAGE_EMPTY_REC);
+            } else {
+                lineRec = Integer.parseInt(argMultimap.getAllValues(PREFIX_LINE_NUMBER_RECOMMENDED).get(0));
```

Comment:
Integer.parseInt(argMultimap.getValue(PREFIX_LINE_NUMBER_RECOMMENDED).get())?

Reply:
Oh yes it should be this instead